clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -919,8 +941,10 @@ private void maybeEnsureValid(Record record) {
         }
 
         private void maybeCloseRecordStream() {
-            if (records != null)
+            if (records != null) {
                 records.close();
+                records = null;
```

Comment:
No. I think I put it in when debugging a test failure and then forgot to remove it.